pkg/mimirpb/prealloc_rw2.go

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package mimirpb
+
+import "fmt"
+
+func ReuseRW2(req *WriteRequest) {
+	reuseSymbolsSlice(req.SymbolsRW2)
+}
+
+// FromWriteRequestToRW2Request converts a write request with RW1 fields populated to a write request with RW2 fields populated.
+// It makes a new RW2 request, leaving the original request alone - it is still up to the caller to free the provided request.
```

Comment:
[nit] Clarify that the some references from the input request are preserved, so it's not safe to free / reuse the input the request data structures until the returned request is still in use.